nixos/modules/services/continuous-integration/gitlab-runner.nix

```diff
@@ -4,14 +4,76 @@ with lib;
 let
   cfg = config.services.gitlab-runner;
   hasDocker = config.virtualisation.docker.enable;
-  hashedServices = mapAttrs'
-    (name: service: nameValuePair
-      "${name}_${config.networking.hostName}_${
-        substring 0 12
-        (hashString "md5" (unsafeDiscardStringContext (toJSON service)))}"
-      service)
-    cfg.services;
+  hashedServices = mapAttrsToList
+    (name: service: "${name}:${hashString "md5" (unsafeDiscardStringContext (toJSON service))}")
+     cfg.services;
   configPath = "$HOME/.gitlab-runner/config.toml";
+  getRunnerServices = (pkgs.writers.writePython3 "get-runner-services" {
+    libraries = with pkgs.python3Packages; [ click ];
+  } ''
+    import click
```

Comment:
we can achieve a similar friendliness with just argparse. Better not to depend on external package.